p2p/src/peer_manager/mod.rs

```diff
@@ -677,19 +678,25 @@ where
         // Expired banned addresses are dropped here, keep this call!
         self.peerdb.heartbeat();
 
-        let pending_outbound = self.pending_outbound_connects.keys().cloned().collect();
-        let connected_outbound_count = self
+        // Do not take into account reserved nodes
+        let pending_outbound = self
+            .pending_outbound_connects
+            .keys()
+            .filter(|addr| !self.peerdb.is_reserved_node(addr))
+            .cloned()
+            .collect::<BTreeSet<_>>();
+        let connected_outbound = self
             .peers
             .values()
             .filter(|peer| {
-                // Do not take into account reserved nodes
                 peer.role == Role::Outbound && !self.peerdb.is_reserved_node(&peer.address)
             })
-            .count();
+            .map(|peer| peer.address.clone())
+            .collect::<BTreeSet<_>>();
+        let all_outbound =
+            pending_outbound.union(&connected_outbound).cloned().collect::<BTreeSet<_>>();
```

Comment:
You can skip the intermediate collects to BTreeSet + union, by chaining the iterators and collecting once.

```
let pending_outbound_iter = self
                .pending_outbound_connects
                .keys()
                .filter(|addr| !self.peerdb.is_reserved_node(addr))
                .cloned();
let connected_outbound_iter = self
    .peers
    .values()
    .filter(|peer| {
        peer.role == Role::Outbound && !self.peerdb.is_reserved_node(&peer.address)
    })
    .map(|peer| peer.address.clone());

let all_outbound = pending_outbound_iter
        .chain(connected_outbound_iter)
        .collect::<BTreeSet<_>>();
```

Reply:
Right, thanks, updated.